### PR TITLE
Fixing $this call on closure for still be PHP 5.3.x compatible.

### PR DESCRIPTION
**Pull request by brunoric**
Since in composer.json the minimum version required of PHP is 5.3.x, people are using it with that version. This pull request fix the issue #16. Please note that the pull request #19 also fix other issue related with CurlRemoteFilesystem.php file.

### CODE REVIEW

**Comment by hirak** [comment]
Yes, you are right.
But this code isn't work well.... Because `CurlRemoteFilesystem::exec` is a private function.

File: src/CurlRemoteFilesystem.php [hunk 1]
```diff
@@ -88,12 +88,13 @@ public function copy($origin, $fileUrl, $fileName, $progress=true, $options=arra
      */
     public function getContents($origin, $fileUrl, $progress=true, $options=array())
     {
-        return $this->fetch($origin, $fileUrl, $progress, $options, function($ch, $request){
+        $obj = $this;
+        return $this->fetch($origin, $fileUrl, $progress, $options, function($ch, $request, &$obj){
```
Comment (hirak) [inline src/CurlRemoteFilesystem.php]:
```php
... function($ch, $request) use(&$obj) {
```

**Comment by brunoric** [comment]
@hirak How about now? I commited the fix with parent scope reuse (with `use` keyword).

**Comment by hirak** [comment]
@brunoric Thank you. But it still doesn't work.
I found another bug at PHP5.3. I'll rewrite at #22

**Comment by brunoric** [comment]
No problem @hirak. I see the changes that you did. :+1:

**Pull request closed without merging**